core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -329,6 +329,18 @@ sealed trait ZIO[-R, +E, +A]
   ): ZIO[R1, E1, A1] =
     catchSomeDefect { case t => h(t) }
 
+  /**
+   * Recovers from all failures with provided function.
+   *
+   * {{{
+   * effect.catchAllFailure(_ => backup())
+   * }}}
+   */
+  final def catchAllFailure[R1 <: R, E1 >: E, A1 >: A](h: Cause.Fail[E] => ZIO[R1, E1, A1])(implicit
```

Comment:
2 comments here:
1. This is quite misleading because it's not catching all failures, just the first one in the cause. e.g., in the case of `ZIO.fail("boom") &> ZIO.fail("ouch")` only one of the failures will be returned.
2. We should add an `implicit ev: CanFail[E]` requirement so that this method can only be called when the effect can fail

Reply:
regarding 1, I see your point. Anyway, this is following the same convention as `catchAllDefects`. Which also only catches the first defect.

Reply:
I fixed both 1 and 2. For 1, Cause[E] will be provided with all failures.